[4.1] TinyMCE changes for child templates 3/3

dgrammatiko opened the pull request:
Pull Request for Issue # .

### Summary of Changes
- Copy the needed methods from HTMLHelper in the tinyMCE so that the `editor.css`  is included correctly for the active front end template

### Testing Instructions

Apply the patch in the current 4.1
Verify that the editor.css is loading as expected

Apply #35874 and then this PR
Verify that the editor.css is loading as expected

### Actual result BEFORE applying this Pull Request



### Expected result AFTER applying this Pull Request

No debug:
![Screenshot 2021-11-10 at 01 26 14](https://user-images.githubusercontent.com/3889375/141027112-f354fe60-36f6-4104-bbef-e8ddc73b95aa.png)

With debug enabled:
![Screenshot 2021-11-10 at 01 27 30](https://user-images.githubusercontent.com/3889375/141027237-d34ddc71-f27a-4b24-9877-247f37e5749d.png)


### Documentation Changes Required

No

dgrammatiko committed:
Alterbative

dgrammatiko committed:
CS

PhilETaylor commented on plugins/editors/tinymce/tinymce.php:
```diff
@@ -1127,4 +1085,329 @@ private function removeElementWithValue($array, $key, $value)
 		return $array;
 	}
+
+	/**
+	 * Helper function to get the active Site template
+	 *
+	 *
+	 * @return  object|null
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
```
```suggestion
	/**
	 * Helper function to get the active Site template
	 *
	 * @return  object|null
	 *
	 * @since   __DEPLOY_VERSION__
	 */
```

dgrammatiko committed:
Update plugins/editors/tinymce/tinymce.php

Co-authored-by: Phil E. Taylor <phil@phil-taylor.com>

PhilETaylor commented on plugins/editors/tinymce/tinymce.php:
```diff
@@ -1127,4 +1085,328 @@ private function removeElementWithValue($array, $key, $value)
+	/**
+	 * Helper function to get the active Site template
+	 *
+	 * @return  object|null
```
Factually incorrect. Line 1120 returns a string.

dgrammatiko replied:
fixed

PhilETaylor commented on plugins/editors/tinymce/tinymce.php:
```diff
@@ -1127,4 +1085,328 @@ private function removeElementWithValue($array, $key, $value)
+
+							if (!empty($found))
+							{
+								$includes[] = $found;
```
Here we are FOUR if statements down. Is there anyway to refactor to return early to avoid excessive nesting of if statements.

dgrammatiko replied:
@PhilETaylor this method is an almost identical copy of https://github.com/joomla/joomla-cms/blob/a94decccac55949997c6fd0227b26abfc5457fec/libraries/src/HTML/HTMLHelper.php#L382-L655

I just removed the parts that weren't relevant here...

dgrammatiko committed:
string not null

dgrammatiko committed:
Fallback if the template selection fails

dgrammatiko committed:
Merge branch '4.1-dev' into 4.1-dev-tinyMCE-changes-2

bembelimen committed:
Merge branch '4.1-dev' into 4.1-dev-tinyMCE-changes-2

dgrammatiko committed:
Merge branch '4.1-dev' into 4.1-dev-tinyMCE-changes-2

dgrammatiko committed:
Redo

- simplify the logic for the editor.css
- add the code from #35669 for the template snippets to support files placed in the media folder (aka publically accessible)

PhilETaylor commented on plugins/editors/tinymce/src/Field/TemplateslistField.php:
```diff
@@ -0,0 +1,100 @@
+<?php
+
```
```suggestion
```

PhilETaylor commented on plugins/editors/tinymce/src/Field/TemplateslistField.php:
```diff
@@ -0,0 +1,100 @@
+ * Generates the list of directories  available for drag and drop upload.
+ *
+ * @package     Joomla.Plugin
+ * @subpackage  Editors.tinymce
```
```suggestion
```

PhilETaylor commented on plugins/editors/tinymce/src/Field/TemplateslistField.php:
```diff
@@ -0,0 +1,100 @@
+    $options     = [0 => $def];
+    $directories = [JPATH_ROOT . '/templates', JPATH_ROOT . '/media/templates/site'];
+
+    foreach ($directories as $directory) {
+      $this->directory = $directory;
+      $options         = array_merge($options, parent::getOptions());
+    }
```
```suggestion
    foreach ($directories as $directory) 
    {
      $this->directory = $directory;
      $options         = array_merge($options, parent::getOptions());
    }
```

PhilETaylor commented on plugins/editors/tinymce/tinymce.php:
```diff
@@ -1164,287 +1153,88 @@ protected function getActiveSiteTemplate()
-		if (! (array) $template)
-		{
-			return Uri::root(true) . 'media/system/css/editor.css';
+		if (!(array) $template) {
+			return $fallback;
```
```suggestion
		if (!(array) $template) 
		{
			return $fallback;
```

PhilETaylor commented on plugins/editors/tinymce/tinymce.php:
```diff
@@ -1164,287 +1153,88 @@ protected function getActiveSiteTemplate()
 		// We are handling a name.min.ext file:
-		if ($position !== false)
-		{
+		if ($position !== false) {
```
```suggestion
		if ($position !== false)
		{
```

PhilETaylor commented on plugins/editors/tinymce/tinymce.php:
```diff
@@ -1164,287 +1153,88 @@ protected function getActiveSiteTemplate()
 			$nonMinifiedPath = substr_replace($path, '', $position, 4);
 
-			if ($debugMode)
+			if (JDEBUG && is_file($nonMinifiedPath)) {
```
```suggestion
			if (JDEBUG && is_file($nonMinifiedPath))
			{
```

PhilETaylor commented:
This branch is out-of-date with the base branch - you need to merge 4.1-dev into it again

dgrammatiko committed:
Update plugins/editors/tinymce/src/Field/TemplateslistField.php

Co-authored-by: Phil E. Taylor <phil@phil-taylor.com>

dgrammatiko committed:
Update plugins/editors/tinymce/src/Field/TemplateslistField.php

Co-authored-by: Phil E. Taylor <phil@phil-taylor.com>

dgrammatiko committed:
Update plugins/editors/tinymce/src/Field/TemplateslistField.php

Co-authored-by: Phil E. Taylor <phil@phil-taylor.com>

dgrammatiko committed:
Update plugins/editors/tinymce/tinymce.php

Co-authored-by: Phil E. Taylor <phil@phil-taylor.com>

dgrammatiko committed:
Update plugins/editors/tinymce/tinymce.php

Co-authored-by: Phil E. Taylor <phil@phil-taylor.com>

dgrammatiko committed:
Update plugins/editors/tinymce/tinymce.php

Co-authored-by: Phil E. Taylor <phil@phil-taylor.com>

dgrammatiko committed:
Merge branch '4.1-dev' into 4.1-dev-tinyMCE-changes-2

dgrammatiko committed:
description

dgrammatiko committed:
Merge branch '4.1-dev' into 4.1-dev-tinyMCE-changes-2

dgrammatiko committed:
Merge branch '4.1-dev' into 4.1-dev-tinyMCE-changes-2

dgrammatiko commented:
This has been moved to #35998
